### ISSUE TITLE
[Question]: xcode@5 is not picking the entitlement file when submitting archive to Apple TestFlight

### ISSUE THREAD

**Issue by mustafaibrahim989** [microsoft/azure-pipelines-tasks#19005]
### Task name

xcode@5

### Task version

5

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

macOS-13

### Question

```shell
xcode@5 is not picking the entitlement file when submitting archive to Apple TestFlight

The IPA received without any errors to TestFlight but I got ITMS-90078: Missing Push Notification Entitlement after the processing completes. 

Here's a sample of my pipeline

- task: Xcode@5
          inputs:
            sdk: 'iphoneos'
            actions: 'archive'
            packageApp: true
            exportOptions: 'plist'
            scheme: ${{ parameters.scheme }}
            configuration: ${{ parameters.configuration }}
            exportOptionsPlist: ${{ parameters.exportOptionsPlist }}
```

**Comment by aleksandrlevochkin** [comment]
Hi @mustafaibrahim989, thank you for reaching out regarding this issue! We are working on more prioritized issues at the moment but will get back to this one soon.

**Comment by rickrl1** [comment]
Has this been confirmed as a bug in Xcode@5? I have the same issue when using Xcode 15 with the macOS-13 vmImage.

**Comment by Wackymax** [comment]
Anyone managed to get this working? I am running into the same problem

**Comment by MWKhalid** [comment]
Hi @aleksandrlevochkin , any updates on this, or any workaround you might want to share with us, please?

**Comment by alvarosg1997** [comment]
> Hi @aleksandrlevochkin , any updates on this, or any workaround you might want to share with us, please?

Hi! I have the same problem, have you been able to solve it?

**Comment by lettucemode** [comment]
We're having a similar issue trying to get the associated-domains entitlement to be picked up by the Xcode build task. The resulting .ipa is always missing it, but when the same files are built on a developer's machine, there are no issues. We've tried dozens of variations on parameters & arguments to no avail. Any guidance or documentation would be much appreciated.

**Comment by efratyos** [comment]
I'm facing the same. Broke my head over this. 
Can someone please share if he got it working?

**Comment by pankti28** [comment]
@efratyos  @lettucemode @alvarosg1997 I'm facing the same issue, anyone managed to get this working ?

**Comment by lettucemode** [comment]
@pankti28 no, we ended up switching to fastlane and doing our build through that instead. It took a day or three to configure what we wanted, but it works great.

**Comment by pankti28** [comment]
@lettucemode the issue is fixed for me, i just added new build task with xcode@5 before archive.

**Comment by lettucemode** [comment]
@pankti28 do you mean, you have two separate XCode@5 tasks, one that does the build action and one that does the archive action?

**Comment by pankti28** [comment]
@lettucemode that's correct, and provide archivepath there
  - task: Xcode@5
    displayName: 'Build IPA'
    inputs:
      actions: 'clean build'
      configuration: 'Release-QA' #change with your release configuration
      sdk: 'iphoneos'
      xcWorkspacePath: 'ios/Runner.xcworkspace'
      workingDirectory: '$(Build.SourcesDirectory)'
      scheme: 'QA' #change with your scheme name that you have in xcode
      packageApp: true
      signingOption: 'manual'
      signingIdentity: '$(APPLE_CERTIFICATE_SIGNING_IDENTITY)'
      provisioningProfileName: '$(APPLE_PROV_PROFILE_UUID)'
      archivePath: 'build/ios/iphoneos/Runner.xcarchive'

**Comment by mattbroussard** [comment]
In case it is of use to anyone, I had a similar issue and I resolved it by adding a separate task at the end using Fastlane's "[resign](https://docs.fastlane.tools/actions/resign/)" action to re-sign the IPA output by the Xcode@5 task with the correct entitlements file (this tool is [already present](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#tools) in the runner image we're using):
```yaml
- bash: |
    fastlane sigh resign \
      $(build.artifactstagingdirectory)/AppName.ipa \
      --entitlements "$(EntitlementsPath)" \
      --signing_identity "$(SigningIdentityName)" \
      --provisioning_profile "$(ProvisioningProfilePath)"
  displayName: Re-sign IPA with Entitlements
```

This is roughly in-line with what the previous few comments are suggesting, but accomplished in a slightly different way.

From what I've gathered from various threads, my understanding is that the problem stems from two factors:
- `xcodebuild`'s export phase signing does not consider the entitlements file; entitlements must be included in a signature done earlier (e.g. at archive phase as suggested above). But Xcode@5 by default (unless `signingOption` is specified) passes CODE_SIGNING_ALLOWED=NO to the earlier phases and only allows signing during export.
- Some projects, like ours, cannot use `signingOption: manual` to sign earlier in the process because Xcode@5 sets the provisioning profile in a way that applies to frameworks that don't support provisioning profiles (see [here](https://developercommunity.visualstudio.com/t/Not-able-to-sign-manually-for-applestore/1301176#T-N1303891)).
